routes/linkUrls.js

```diff
@@ -0,0 +1,33 @@
+const express = require('express');
+const router = express.Router();
+const cache = require('memory-cache');
+const asyncHandler = require('express-async-handler');
+
+const handleCache = require('../helpers/handleCache');
+const commonContent = require('../helpers/commonContent');
+
+router.get('/:codenames', asyncHandler(async (req, res, next) => {
+  await handleCache.evaluateCommon(res, ['urlMap']);
+
+  const KCDetails = commonContent.getKCDetails(res);
+  const urlMap = await cache.get(`urlMap_${KCDetails.projectid}`);
+  const codenames = req.params.codenames.split('/');
+
+  if (codenames.length > 0) {
```

Comment:
I would expect this kind of check to be the first thing that happens in the method…
```suggestion
  const codenames = req.params.codenames.split('/');
  if (codenames.length <= 0) {
     return;
  }
```
But shouldn't you call next in the scenario above? Or return `404`? (i.e. when there is no code name provided)